features/git-sync/all_branches/feature_branch_merge_main_branch_conflict/without_remote_origin/first_branch/with_open_changes.feature

```diff
@@ -0,0 +1,122 @@
+    And the following commits exist in my repository
+      | BRANCH   | LOCATION | MESSAGE         | FILE NAME        | FILE CONTENT     |
+      | main     | local    | main commit     | conflicting_file | main content     |
+      | feature1 | local    | feature1 commit | conflicting_file | feature1 content |
+      | feature2 | local    | feature2 commit | feature2_file    | feature2 content |
+    And I am on the "main" branch
+    And I have an uncommitted file with name: "uncommitted" and content: "stuff"
+    When I run `git sync --all`
+
+
+  @finishes-with-non-empty-stash
+  Scenario: result
+    Then it runs the Git commands
+      | BRANCH   | COMMAND                  |
+      | main     | git stash -u             |
+      | main     | git checkout feature1    |
+      | feature1 | git merge --no-edit main |
+    And I get the error
+      """
+      To abort, run "git sync --abort".
+      To continue after you have resolved the conflicts, run "git sync --continue".
+      To skip the sync of the 'feature1' branch, run "git sync --skip".
+      """
+    And I end up on the "feature1" branch
+    And I don't have an uncommitted file with name: "uncommitted"
+    And my repo has a merge in progress
+
+
+  Scenario: aborting
+    When I run `git sync --abort`
+    Then it runs the Git commands
+      | BRANCH   | COMMAND           |
+      | feature1 | git merge --abort |
+      | feature1 | git checkout main |
+      | main     | git stash pop     |
+    And I end up on the "main" branch
+    And I again have an uncommitted file with name: "uncommitted" and content: "stuff"
+    And I have the following commits
+      | BRANCH   | LOCATION | MESSAGE         | FILE NAME        |
+      | main     | local    | main commit     | conflicting_file |
+      | feature1 | local    | feature1 commit | conflicting_file |
+      | feature2 | local    | feature2 commit | feature2_file    |
+
+
+  Scenario: skipping
+    When I run `git sync --skip`
+    Then it runs the Git commands
+      | BRANCH   | COMMAND                  |
+      | feature1 | git merge --abort        |
+      | feature1 | git checkout feature2    |
+      | feature2 | git merge --no-edit main |
+      | feature2 | git checkout main        |
+      | main     | git stash pop            |
+    And I end up on the "main" branch
+    And I again have an uncommitted file with name: "uncommitted" and content: "stuff"
+    And I have the following commits
+      | BRANCH   | LOCATION | MESSAGE                           | FILE NAME        |
+      | main     | local    | main commit                       | conflicting_file |
+      | feature1 | local    | feature1 commit                   | conflicting_file |
+      | feature2 | local    | feature2 commit                   | feature2_file    |
+      |          |          | main commit                       | conflicting_file |
+      |          |          | Merge branch 'main' into feature2 |                  |
+
```

Comment:
You should add verification of the files here, so that it is clear to the reader how the file system looks like when they skip certain operations. Here is an example:

```cucumber
    And now I have the following committed files
      | BRANCH   | NAME             | CONTENT          |
      | main     | conflicting_file | main content     |
      | feature1 | conflicting_file | feature1 content |
      | feature2 | feature2_file    | feature2 content |
```